modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/DXPActivationKeysTable/components/BadgeFilter/index.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {useCallback} from 'react';
+import Button from '../../../../../../common/components/Button';
+import getCurrentEndDate from '../../../../../../common/utils/getCurrentEndDate';
+import BadgePillFilter from '../../../../components/BadgePillFilter';
+import {INITIAL_FILTER} from '../../utils/constants/initialFilter';
+
+const BadgeFilter = ({
+	activationKeysLength,
+	loading,
+	filtersState: [filters, setFilters],
+}) => {
+	const getDatesDisplay = useCallback((dateFilterState) => {
+		const dateDisplays = [];
+
+		if (dateFilterState.value?.onOrAfter) {
+			const todayDNE = new Date();
+			todayDNE.setFullYear(todayDNE.getFullYear() + 100);
+
+			if (new Date(dateFilterState.value?.onOrAfter) >= todayDNE) {
+				return 'DNE';
+			}
+
+			dateDisplays.push(
+				getCurrentEndDate(dateFilterState.value?.onOrAfter)
+			);
+		}
+
+		if (dateFilterState.value?.onOrBefore) {
+			dateDisplays.push(
+				getCurrentEndDate(dateFilterState.value?.onOrBefore)
+			);
+		}
+
+		return dateDisplays.join(' – ');
+	}, []);
+
+	const getKeyTypeDisplay = useCallback(
+		(filterKeyType) => {
+			const keyTypesDisplay = [];
+
+			if (filterKeyType.value?.hasOnPremise) {
+				keyTypesDisplay.push('On-Premise');
+			}
+
+			if (filterKeyType.value?.hasVirtualCluster) {
+				if (
+					!(
+						filterKeyType.value?.minNodes ||
+						filterKeyType.value?.maxNodes
+					)
+				) {
+					keyTypesDisplay.push(`Virtual Cluster`);
+				}
+				else if (
+					filterKeyType.value?.minNodes ===
+					filterKeyType.value?.maxNodes
+				) {
+					keyTypesDisplay.push(
+						`Virtual Cluster (${filterKeyType.value?.minNodes} nodes)`
+					);
+				}
+				else {
+					const nodesDisplay = [];
+
+					if (filterKeyType.value?.minNodes) {
+						nodesDisplay.push(filterKeyType.value?.minNodes);
+					}
+
+					if (filterKeyType.value?.maxNodes) {
+						nodesDisplay.push(filterKeyType.value?.maxNodes);
+					}
+
+					keyTypesDisplay.push(
+						`Virtual Cluster (${nodesDisplay.join('-')} nodes)`
+					);
+				}
+			}
+
+			return (
+				<BadgePillFilter
+					filterName={filterKeyType.name}
+					filterValue={keyTypesDisplay.join(', ')}
+					onClick={() =>
+						setFilters((previousFilters) => ({
+							...previousFilters,
+							keyType: {
+								...previousFilters.keyType,
+								value: {
+									hasOnPremise: undefined,
+									hasVirtualCluster: undefined,
```

Comment:
prefer using false against undefined
```suggestion
									hasOnPremise: false,
									hasVirtualCluster: false,
```

Reply:
done!